tt_metal/impl/dispatch/kernels/cq_prefetch.cpp

```diff
@@ -117,6 +120,11 @@ void read_from_pcie(volatile tt_l1_ptr uint32_t *& prefetch_q_rd_ptr,
                     uint32_t cmd_ptr,
                     uint32_t size) {
 
+    // When stalled, prevent prefetcher from fetching next cmd via early exit.
+    if (stall_state == STALLED) {
```

Comment:
I'm thinking we want this test in get_cmds so that we don't even read the fetch_q when stalled

Reply:
Okay cool yes I agree, this version you commented on was outdated, and the version after the followup commit has it in fetch_q_get_cmds(), will keep it there.